### PR TITLE
add vk emulator

### PR DESCRIPTION
**Pull request by ksyashka**

### CODE REVIEW

File: src/main/java/com/incuube/copycat/domain/vk/VkStatus.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	OK,
+	NOT_ENOUGHT_DATA,
+	INCORRECT_SIGNATURE,
+	ERROR,
+	UNSUPPORTED_NUMBER,
+	INCORRECT_NUMBER,
+	NUMBER_IN_BLACK_LIST,
+	NUMBER_TYPE_NOT_ALLOWED,
+	RATELIMIT,
+	UNSUPPORTED_TEMPALTE
```
Comment (WladekK) [inline src/main/java/com/incuube/copycat/domain/vk/VkStatus.java]: template

File: src/main/java/com/incuube/copycat/constants/CommonConstants.java [hunk 2]
```diff
@@ -6,6 +6,12 @@
 	public static final String TRANSACTION = "transaction";
 	public static final String PROMOTION = "promotion";
 
+	public static final int MILLIS_IN_SECOND = 1000;
+	//HTTP Sender
+	public static final String DEFAULT_CONTENT_TYPE = "application/json";
+	public static final int DEFAULT_TTL_HTTP_SENDER = 40;
```
Comment (dmytr0) [inline src/main/java/com/incuube/copycat/constants/CommonConstants.java]: why default ttl 40?
is it in seconds?

File: src/main/java/com/incuube/copycat/http/sender/HttpSender.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+import static com.incuube.copycat.constants.CommonConstants.DEFAULT_CONTENT_TYPE;
+import static com.incuube.copycat.constants.CommonConstants.DEFAULT_TTL_HTTP_SENDER;
+import static com.incuube.copycat.constants.CommonConstants.MILLIS_IN_SECOND;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+public class HttpSender {
+
```
Comment (dmytr0) [inline src/main/java/com/incuube/copycat/http/sender/HttpSender.java]: remake to FutureRequestExecutionService

File: src/main/java/com/incuube/copycat/http/sender/VkDrSender.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+			vkDrExecutionService.execute(httpUriRequest, null, response -> response, retryHandler);
+			logger.debug("Vk DR was sent dr to " + httpWithParams);
+		} catch (Exception e) {
+			logger.warn("Cannot send vk dr: ", e);
+		}
+	}
+
+	private String urlWithParams(String url, Map<String, String> params) throws URISyntaxException {
+		URIBuilder builder = new URIBuilder(url);
+		List<NameValuePair> nvps = new ArrayList<>();
```
Comment (dmytr0) [inline src/main/java/com/incuube/copycat/http/sender/VkDrSender.java]: nvps -> requestParams

File: src/main/java/com/incuube/copycat/config/ServiceConfig.java [hunk 5]
```diff
@@ -15,6 +16,7 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.annotation.EnableScheduling;
```
Comment (dmytr0) [inline src/main/java/com/incuube/copycat/config/ServiceConfig.java]: redundant import?

File: src/main/java/com/incuube/copycat/service/vk/VkCommonValidator.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		if (vkRequest.getSignature() == null) {
+			return VkResponse.builder().status(VkStatus.ERROR).detailStatus("INCORRECT_SIGNATURE").build();
+		}
+
+		if (vkRequest.getPhone() == null) {
+			return VkResponse.builder().status(VkStatus.ERROR).detailStatus("UNDEFINED_PHONE_NUMBER").build();
+		}
+		int ttl = vkRequest.getTtl();
+		if (ttl < 60 || ttl > 86400) {
+			return VkResponse.builder().status(VkStatus.ERROR).detailStatus("INCORRECT_TTL").build();
```
Comment (dmytr0) [inline src/main/java/com/incuube/copycat/service/vk/VkCommonValidator.java]: why status the same

Comment (ksyashka) [reply src/main/java/com/incuube/copycat/service/vk/VkCommonValidator.java]: mailru return the same

**Pull request merged**